app: http: Add hex_format option to AT#XHTTPCREQ (#323)
nrfconnect/ncs-serial-modem: doc/app/at_httpc.rst

::

>    AT#XHTTPCREQ=<handle>,<url>,<method>[,<auto_reception>[,<hex_format>[,<body_len>[,<header 1>[,<header 2>[...]]]]]]

Reviewer: Should we rename `hex_format` to `format` to make this more future proof? Same for CoAP.

   #XHTTPCSTAT: 0,200,432,0

> HTTP GET with Range header (``hex_format=0`` and ``body_len=0`` are required as placeholders when ``<header>`` parameters follow):

Reviewer 2: ```suggestion
HTTP GET data in binary format with Range header (``body_len=0`` is required as a placeholder when ``<header>`` parameters follow):
```
